anthropic-java-bedrock/src/main/kotlin/com/anthropic/bedrock/backends/BedrockBackendAdapter.kt

```diff
@@ -0,0 +1,592 @@
+                        cachedBytes = buffer.toByteArray()
+                        return cachedBytes!!
+                    } catch (e: Exception) {
+                        throw AnthropicException("Error writing request", e)
+                    }
+                }
+
+                override fun writeTo(outputStream: OutputStream) {
+                    outputStream.write(serialize())
+                }
+
+                override fun contentType(): String = CONTENT_TYPE_JSON
+
+                override fun contentLength(): Long {
+                    return serialize().size.toLong()
+                }
+
+                override fun repeatable(): Boolean = true
+
+                override fun close() {}
+            }
+        }
+
+        /**
+         * Creates a JSON [ObjectNode] representing the JSON data parsed from a
+         * [HttpRequestBody].
+         *
+         * @param body A new request body holding the JSON data.
+         * @param jsonMapper The mapper to use to deserialize the JSON data.
+         *
+         * @return The JSON object data to form the request body, or `null` if
+         *     the request does not have a body or if the body is empty.
+         */
+        // NOTE: "internal" visibility allows this method to be used in unit
+        // tests to create test fixtures.
+        @JvmSynthetic
+        internal fun bodyToJson(
+                body: HttpRequestBody?, jsonMapper: ObjectMapper): ObjectNode? {
+            val jsonData = ByteArrayOutputStream()
+
+            body?.writeTo(jsonData)
+            if (jsonData.size() > 0) {
+                return jsonMapper.readValue(
+                    jsonData.toByteArray(), ObjectNode::class.java)
+            }
+            return null
+        }
+    }
+
+    /**
+     * Gets the base URL for the Amazon Bedrock runtime service. The base URL
+     * includes the Bedrock runtime service name and the AWS region.
+     */
+    override fun baseUrl(): String =
+        // Could use the AWS "BedrockEndpointProvider" in "fromEnv()", but that
+        // is a large dependency that brings in all the competing AWS SDK
+        // classes for Bedrock that are not desired in this SDK.
+        "https://$SERVICE_NAME.$region.amazonaws.com"
+
+    /**
+     * Prepares the request for use with Amazon Bedrock.
+     *
+     * A number of changes are made to support the requirements of AWS requests
+     * to the Bedrock backend:
+     *
+     * * The model ID is moved from the JSON body content to the path segments.
+     * * The AWS action to be performed is added to the path segments.
+     * * The Anthropic version is added to the JSON body content.
+     * * The Anthropic beta versions (if present) are copied from the headers
+     *     to the JSON body content. Beta versions are specified in (optional)
+     *     `anthropic-beta` headers. There can be more than one header with more
+     *     than one version per header (comma-separated).
```

Comment:
Should this part be column-aligned with the "The" in "The Anthropic beta versions" or is this just what the formatter did